src/libsnark/algebra/fields/bigint.tcc

```diff
@@ -201,7 +201,7 @@ inline bigint<m> bigint<n>::shorten(const bigint<m>& q, const char *msg) const
         }
     }
     bigint<m> res;
-    mpn_copyi(res.data, data, n);
+    mpn_copyi(res.data, data, m);
```

Comment:
I think this fix should be in a separate commit. (We decided that it was probably not exploitable in Zcash, so it's OK for the commit to give more details the bug.)